File: example/cocoapods/ExampleUITests/ExampleUITests.m
//
//  ExampleUITests.m
//  ExampleUITests
//
//  Created by Felix Krause on 30/07/15.
//  Copyright © 2015 Felix Krause. All rights reserved.
//

#import <XCTest/XCTest.h>

@interface ExampleUITests : XCTestCase

@end

@implementation ExampleUITests

- (void)setUp {
    [super setUp];
    
    // Put setup code here. This method is called before the invocation of each test method in the class.
    
    // In UI tests it is usually best to stop immediately when a failure occurs.
    self.continueAfterFailure = NO;
    // UI tests must launch the application that they test. Doing this in setup will make sure it happens for each test method.
    [[[XCUIApplication alloc] init] launch];
}

- (void)tearDown {
    // Put teardown code here. This method is called after the invocation of each test method in the class.
    [super tearDown];
}

- (void)testExample {
    // Use recording to get started writing UI tests.
    // Use XCTAssert and related functions to verify your tests produce the correct results.
}

@end

File: example/ExampleUITests/ExampleUITests.swift
//
//  ExampleUITests.swift
//  ExampleUITests
//
//  Created by Felix Krause on 19/06/15.
//  Copyright © 2015 Felix Krause. All rights reserved.
//

import Foundation
import XCTest

class ExampleUITests: XCTestCase {
        
    override func setUp() {
        super.setUp()
        
        let app = XCUIApplication()
        setupSnapshot(app)
        app.launch()
    }
    
    func testExample()
    {
        snapshot("0Launch")
        let tabBar = XCUIApplication().tabBars
        let secondButton = tabBar.buttons.elementBoundByIndex(1)

        XCUIDevice().orientation = UIDeviceOrientation.LandscapeLeft
        snapshot("1LandscapeLeft")
        
        secondButton.tap()
        XCUIDevice().orientation = UIDeviceOrientation.LandscapeRight
        snapshot("2LandscapeRight")

        secondButton.tap()
        XCUIDevice().orientation = UIDeviceOrientation.Portrait
        snapshot("3Portrait")
    }
}
